javascript/javascript3/week3/paintcircle/shoppingcart/shoppingcart.js

```diff
@@ -0,0 +1,156 @@
+
+                const li1 = document.createElement('li');//
+                h5.innerHTML = `fullname of the repo - ${data.items[0].full_name}`;
+
+                li1.innerHTML = `url of the repo - ${data.items[0].url}`;
+
+                const li2 = document.createElement('li');
+                li2.innerHTML = `owner of the repo - ${data.items[0].owner.login}`;
+
+
+                ul.append(h5, li1, li2)
+            })
+
+            )
+
+    }
+}
+
+
+const use = new respiratory("Hailegebriel22", "Hanouj", "patel-prakashkumar")
+
+
+
+
+
+
+// Shopping cart using Classes
+
+class Product {
+    constructor(name, price) {
+        this.name = name;
+        this.price = price;
+    }
+    //provided currency return the correct price 
+    convertToCurrency(currency) {
+        const currencyValue = fetch(`https://v6.exchangerate-api.com/v6/ce21fbeaedb5d8ad0bf28163/latest/DKK`)
+            .then(data => data.json());
+        if (currency === "USD") {
+
+            currencyValue.then(datar => console.log(this.price * datar.conversion_rates.USD))
+        }
+        else if (currency === "EUR") {
+            currencyValue.then(datar => console.log(this.price * datar.conversion_rates.EUR))
+        }
+        else if (currency === "GBP") {
+            currencyValue.then(datar => console.log(this.price * datar.conversion_rates.GBP))
+        }
+
+    }
+}
+
+class ShoppingCart {
+    constructor() {
+        this.products = [];
+    }
+
+    addProduct(product) {
+        return this.products.push(product);
+    }
+
+    removeProduct(product) {
+        return this.products.filter(item => item !== product);
+    }
+
+    searchProduct(productName) {
+        const filteredProducts = this.products.filter(product =>
+            product.name.toLowerCase() === productName.toLowerCase())
+        return filteredProducts;
+    }
+
+    getTotal() {
+        const eachPrice = [];
+        this.products.forEach(product => eachPrice.push(product.price));
+        return eachPrice.reduce((x, y) => x + y)
+    }
+
+    renderProducts() {
+        const h1 = document.createElement('h1');
+        document.body.appendChild(h1)
+        h1.innerHTML = `Products price list`
+
+
+        this.products.forEach(product => {
+            const ul = document.createElement('ul');
+            document.body.appendChild(ul)
+
+            const h5 = document.createElement('h5');
+            ul.appendChild(h5)
+
+            const li = document.createElement('li');
+
+            h5.innerHTML = `product - ${product.name}`
+
+            li.innerHTML = `price - ${product.price} kr`
+            //li.style.listStyle = "none"
+
+            return ul.appendChild(li)
+
+        });
+
+    }
+
+    getUser(user) {
+        const dataPromise = fetch(`https://jsonplaceholder.typicode.com/users/${user}`)
+            .then(response => response.json())
+            .then(data => data)
+        return dataPromise;
+
```

Comment:
I don't think getUser works like this, the URL is not the correct URL, check your console.